### PR TITLE
Make command-enter run the query

### LINKED ISSUE #1209: Feature Request: Cmd-enter to run query
Would be nice to be able to run queries while I'm refining queries using a keyboard shortcut

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves #1209

### CODE REVIEW

**Comment by camsaul** [comment]
Quick thought. Does this work for Linux or Windows users?

File: frontend/src/metabase/query_builder/containers/QueryBuilder.jsx [hunk 1]
```diff
@@ -161,19 +163,27 @@ export default class QueryBuilder extends Component {
...
         this.forceUpdateDebounced();
         let viz = ReactDOM.findDOMNode(this.refs.viz);
         if (viz) {
             viz.style.opacity = 0.2;
         }
     }
 
+    handleKeyDown = (e) => {
+        const ENTER_KEY = 13;
+        if (e.keyCode === ENTER_KEY && e.metaKey && this.props.isRunnable) {
```
Comment (camsaul) [inline frontend/src/metabase/query_builder/containers/QueryBuilder.jsx]: I thought Meta is supposed to be the same thing as Alt/Option, right? At least it is in Terminal.app and Emacs

Comment (tlrobinson) [inline frontend/src/metabase/query_builder/containers/QueryBuilder.jsx]: https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/metaKey

> Note: On Macintosh keyboards, this is the command key (⌘). On Windows keyboards, this is the windows key (⊞).

**Comment by tlrobinson** [comment]
> Note: On Macintosh keyboards, this is the command key (⌘). On Windows keyboards, this is the windows key (⊞).

@metabase/core-developers Is this sufficient? Not sure what people expect on Windows

**Comment by salsakran** [comment]
Definitely sufficient. I have no idea what the convention on Windows would be at this point. @camsaul you have any opinions on the "natural" keybinding?

**Comment by camsaul** [comment]
I think on Windows and Linux `Ctrl` instead of super would be more natural. Just like how you do Ctrl-C/Ctrl-V/Ctrl-Z etc. instead of Cmd-C/Cmd-V/Cmd-Z.

You could always enable both so it's easier for people to guess

**Comment by salsakran** [comment]
I'm inclined to pick one. Multiple keybindings to the same thing strike me as too mushy. 
@tlrobinson does it make sense to have OS specific keybindings? If not, I'd selfishly say let's make it work for Mac =)

**Comment by camsaul** [comment]
The Mozilla guide quote seems to suggest that Cmd/Windows/super doesn't work on Linux. It's also worth noting that not all keyboards have a super button 😢

**Comment by tlrobinson** [comment]
Personally I'm fine with sticking with what we have until a Windows/Linux user complains.

OS specific key bindings would be slightly annoying but if @camsaul feels strongly about it I can look into it.

**Comment by salsakran** [comment]
#shipit.

do we want to document or instrument this?

**Comment by mazameli** [comment]
How come we didn't implement this for GUI queries?

**Comment by tlrobinson** [comment]
It seemed a lot more useful for native queries where you're already using the keyboard.

**Comment by salsakran** [comment]
mainly because when you're working with sql, you're using the keyboard, and being able to do cmd+enter prevents you from having to switch to using a mouse with one hand.

in gui mode you're already clicking everywhere, and it doesn't make sense to have you switch to the keyboard to re-run.

**Comment by mazameli** [comment]
The SQL case is obviously useful — I don't see this as an either-or. I think it's weird that it's inconsistently implemented across the product: the run button looks the same in both modes, but it doesn't respond the same. Also, now that we have the ability to commit filters in GUI mode by hitting enter, my hands are in fact often already on the keyboard, and it actually often drives me nuts to have to mouse-click on the run query button when editing filter values. This also totally ignores the fact that lots of people work on laptops where the difference between using the trackpad vs. cmnd+enter is way smaller than in a mouse+keyboard scenario.

**Comment by salsakran** [comment]
No strong opinions on it being an "and".  You asked "how come" and that's what I was answering. 

I have no objection to it also working for gui queries.

**Comment by mazameli** [comment]
Hah, cool. Apologies if I was being too agro. 😂

**Comment by chowchowkelly** [comment]
new to metabase here! Any updates on building a hot key for Mac? If it's already been built, I've tried both command + return and control + return. They didn't work :(

**Pull request merged**